internal/commands/scan.go

```diff
@@ -2492,9 +2486,49 @@ func validateCreateScanFlags(cmd *cobra.Command) error {
 		return errors.Errorf("Invalid value for --project-private-package flag. The value must be true or false.")
 	}
 
+	err = validateThresholds(cmd)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func validateThresholds(cmd *cobra.Command) error {
+	threshold, _ := cmd.Flags().GetString(commonParams.Threshold)
+	if threshold == "" {
+		return nil
+	}
+	threshold = normalizeThresholds(threshold)
+	thresholdLimits := strings.Split(strings.ToLower(threshold), ";")
+	for _, limit := range thresholdLimits {
+		engineName, intLimit, err := parseThresholdLimit(limit)
+		if err != nil {
+			return err
+		}
+		if intLimit < 1 {
+			return errors.Errorf("%s: Threshold limit should be greater or equal to 1\n", engineName)
+		}
+	}
+	return nil
+}
+
+func normalizeThresholds(threshold string) string {
+	return strings.ReplaceAll(strings.ReplaceAll(threshold, " ", ""), ",", ";")
+}
+
+func parseThresholdLimit(limit string) (engineName string, intLimit int, err error) {
+	parts := strings.Split(limit, "=")
+	engineName = strings.Replace(parts[0], commonParams.KicsType, commonParams.IacType, 1)
+	if len(parts) > 1 {
```

Comment:
You can reduce Cyclomatic Complexity if you invert the condition:
if len(parts) <= 1
{
return engineName, intLimit, err
}

and later the logic you currently have inside len(parts) > 1